packages/react/src/components/TreeView/TreeNode.tsx

```diff
@@ -5,14 +5,168 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React, { useState, useEffect, useRef } from 'react';
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  PropsWithChildren,
+  RefObject,
+} from 'react';
 import PropTypes from 'prop-types';
 import { CaretDown } from '@carbon/icons-react';
 import classNames from 'classnames';
 import { keys, match, matches } from '../../internal/keyboard';
 import uniqueId from '../../tools/uniqueId';
 import { usePrefix } from '../../internal/usePrefix';
 
+export interface SelectedNode {
+  id: string | number;
+  label?: React.ReactNode;
+  value: string;
+  activeNodeId?: string | number;
+}
+
+interface TreeNodeProps {
+  /**
+   * Node ID
+   */
+  id: string | number;
+
+  /**
+   * The value of the active node in the tree
+   */
+  active: string | number;
```

Comment:
Can we keep the props `isRequired` values the same as the PropType definitions? Seems like this is enforcing a lot of optional props as required.

Reply:
Absolutely @tw15egan , I've already made the necessary changes to align the `isRequired` values with the PropType definitions. This should no longer enforce optional props as required. Thanks for catching that!